docs/resources/ui-building-blocks/components/user-defined-components/component-lifecycle.md

```diff
@@ -0,0 +1,256 @@
+
+- **Is List:** This property determines whether the variable can hold multiple values of the same
+  data type (like a list or array) or just a single value.
+
+- **Initial Field Value:** This property sets the default value for the variable when it is first
+  created. It's like setting the starting point or the value that the variable begins with before
+  anything else happens.
+
+- **Nullable:** This property determines whether the variable can have a null value. When "*
+  *Nullable**" is set to true, it means the variable can be empty or have a null value. This is
+  useful when dealing with optional data or scenarios where the absence of a value is valid.
+
+Now, let's apply these concepts to the `isFavourite` variable in the context of the above example:
+
+- For the `isFavourite` variable, it is a single value (boolean), so **Is List** would be set to
+  false.
+
+- The **Initial Field Value** would also be set to **false**, indicating that the item is not
+  favorited by default.
+
+- **Nullable** property will be set to false, as the variable should always have a boolean value
+  (true or false) and never be null.
+
+:::note
+You can set the **Data Type** of your Component State variable to any primitive data types such as *
+*String,
+Integer, Boolean, Double** or to any other complex built-in data types such as **Enum, Custom Data
+Type, Document,** etc. To learn more about the available data types, refer the [**Data
+Representation Section.** ](#)
+:::
+
+### Get Component State Value
+
+In the following example, we demonstrate how to toggle the heart icon from an outlined to a filled
+icon based on the `isFavourite` state variable. We introduce a `Conditional Builder` widget that
+allows us to show a widget tree based on **If/Else If/Else** conditions. The goal is to visually
+indicate whether a product has been favorited by the user.
+
+Follow the steps as below:
+
+<div style={{
+    position: 'relative',
+    paddingBottom: 'calc(56.67989417989418% + 41px)', // Keeps the aspect ratio and additional padding
+    height: 0,
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/Y96decdgYWVll3SP9Jk8?embed&show_copy_link=true"
+        title="Get Component State"
+        style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            colorScheme: 'light'
+        }}
+        frameborder="0"
+        loading="lazy"
+        webkitAllowFullScreen
+        mozAllowFullScreen
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+
+### Update Component State [Action]
+
+**Component state** values can only be updated via actions. Whenever you want to update the
+component state, call an action called **Update Component State** from the Action Flow Editor
+of the component.
+
+In the following demo, we open the Action Flow Editor on the parent widget` Conditional Builder` and
```

Comment:
Extra space in code block. should be 'Conditional Builder'

Reply:
Done